source/Shellfish/Windows/Win32Helper.cs

```diff
@@ -1,46 +1,24 @@
 using System;
 using System.ComponentModel;
+using System.Runtime.InteropServices;
 
 namespace Octopus.Shellfish.Windows
 {
     static class Win32Helper
     {
-        public static bool Invoke(Func<bool> nativeMethod, string failureDescription)
+        public static void Invoke(Func<bool> nativeMethod, string failureDescription)
         {
-            try
+            if (!nativeMethod())
             {
-                return nativeMethod() ? true : throw new Win32Exception();
-            }
-            catch (Win32Exception ex)
-            {
-                throw new Exception($"{failureDescription}: {ex.Message}", ex);
+                throw new Exception(failureDescription, new Win32Exception());
```

Comment:
Previously we were throwing and catching a `Win32Exception`, presumably to get the error code and description. Just constructing a `Win32Exception` is enough - we don't need to throw it.

Reply:
This code was fairly terrible; you've improved it, but I feel like we'd be better off without attempting a generic helper.

We call a method which we assume will SetLastError, but then we never call GetLastError and just throw a random generic exception, that's not going to be useful when troubleshooting.

The CLR has a Marshal.ThrowExceptionForWin32Error which solves this properly, but it's nonpublic and/or not available in netstandard :-(

Reply:
(For clarity: I would approve the PR with this code you've written; my idea about removing the generic wrapper is just an idea at this stage, not a blocker)

Reply:
Minor: I believe one behavioural difference here is that the inner exception will no longer have its stacktrace populated. Not a big deal but also not hard to fix:

```suggestion
                throw new Exception(failureDescription, ExceptionDispatchInfo.SetCurrentStackTrace(new Win32Exception()));
```

(what @borland is talking about might be better though!)

Reply:
Good catch. Might as well, I suppose.

Reply:
Scratch that, it's not available in .NET Standard 2.0. All of a sudden the effort has outweighed the value.